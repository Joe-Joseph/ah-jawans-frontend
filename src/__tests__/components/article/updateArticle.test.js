import React from 'react';
import { shallow, mount } from '../../../../config/enzymeConfig';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import UpdateArticle, { UpdateArticle as View } from '../../../components/article/updateArticle';

jest.mock('../../../components/Layout/Layout', () => () => (
  <div id="mockFileLayout">mockFileLayout</div>
));
const mockStore = configureMockStore([thunk]);
const store = mockStore({
  article: { article: {} },
  userCredentials: { userCredentials: {} }, 
});

const props = {
  article: jest.fn(),
  postDataThunkprivate: jest.fn(),
  errors: {}
};

global.MutationObserver = class {
  disconnect() {}
  observe() {}
};

describe('<UpdateArticle />', () => {
  beforeEach(() => { });

  afterEach(() => {
    jest.clearAllMocks();
  });
  const component = shallow(
    <Provider store={store}>
      <UpdateArticle {...props} />
    </Provider>,
  );

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });
 
  it('Should render the component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <UpdateArticle />
        </Router>
      </Provider>,
    );
    expect(wrapper.find('UpdateArticle')).toBeDefined();
  });
test('should test create article', () => {
      const props = {
        title: 'Onboarding onto a large codebase',
        body:'Onboarding onto a large codebase',
        tags:'javascipt, react'
      };
      let instance
        let submitButton
        beforeAll(() => {
            instance = wrapper.instance()
            submitButton = wrapper.find('button[type="submit"]')
          submitButton.simulate('click')
          it('should make a request to the server', () => {
            expect(submitButton).toHaveLength(1)
            wrapper.update()
            const event = {
                preventDefault: jest.fn(),
            }
            wrapper.setProps({
                history: {push: jest.fn()}
            })
            instance.handleSubmit(event)
        })
        })
  });
});
