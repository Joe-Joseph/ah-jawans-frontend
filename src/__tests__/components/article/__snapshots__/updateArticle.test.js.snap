// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateArticle /> should render without crashing 1`] = `ShallowWrapper {}`;
